### DISCUSSION TITLE
Working on Pi 4B

### DISCUSSION THREAD

**Discussion by Mastacarlos** [OlivierGaland/CashFactory#10] [General]
For anyone interested:
I was able to get this working on Pi 4B.  All docker images included in this project except for IPRoyalPawns ad EarnApp do not have arm64 versions of docker. So, you can add an emulator to get them to run. For some reason even though EarnApp has an arm64 docker image it doesn't work on the Pi(at least it didn't for me). So here's what I did to get it all working:

1. Remove or comment EarnApp section  from the docker-compose.yml file
2. Add the following to the docker-compose.yml file
	Emulator:
		image: tonistiigi/binfmt
		privileged: true
		command: -install all
	NOTE: this does not include restart always. It simply needs to be installed and run to work. If you add a restart always
	 then it's status in docker will always show as restarting and Portainer it will show it as not running.
3. Execute this line after completing the rest of the setup
	wget -qO- https://brightdata.com/static/earnapp/install.sh > /tmp/earnapp.sh && sudo bash /tmp/earnapp.sh
	NOTE: this line comes directly from the EarnApp website. It will run as a service instead of a container.
	To regiester simply visit the URL it spits out in a web browswer.

**Comment by Mastacarlos** [comment]
Just realized the formatting won't be correct so make sure you add the indents to match the rest of the YML file otherwise it'll throw errors.

**Comment by OlivierGaland** [comment]
Hello,

Many thanks for your feedback on Pi 4B, I ordered one (with 2Gb RAM) a few weeks ago and I will try your method and check how to integrate it in the project.

Olivier.
